client/service/src/lib.rs

```diff
@@ -149,6 +144,77 @@ impl Spawn for SpawnTaskHandle {
 	}
 }
 
+/// Helper struct to manage background/async tasks in Service.
+pub struct TaskManager {
```

Comment:
Could be moved into its own file.

Reply:
Ok... I think I will improve on api then also, since it won't be internal part of service